PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -248,12 +272,29 @@ def nanoAOD_customizeCommon(process):
     # (it is rerun for run2 within jet tasks while is not needed for newer
     # run3 eras as it is present in miniAOD)
     (run3_nanoAOD_122 | run3_nanoAOD_124).toModify(
-        nanoAOD_tau_switch, runPNetAK4 = True
+        nanoAOD_tau_switch, runUTagCHSAK4 = True
     )
-    nanoAOD_addPNetToTaus(process,
-                          addPNetInfo = nanoAOD_tau_switch.addPNet.value(),
-                          runPNetCHSAK4 = nanoAOD_tau_switch.runPNetAK4.value()
+    
+    # Don't add Unified Tagger for PUPPI jets for Run 2 (as different PUPPI tune
+    # and base jet algorithm) or early Run 3 eras
+    (run3_nanoAOD_122 | run3_nanoAOD_124 | run2_nanoAOD_106Xv2).toModify(
+        nanoAOD_tau_switch, addUParTInfo = False
     )
+    
+    # Add Unified Tagger For CHS Jets (PNet 2023)
+    nanoAOD_addUTagToTaus(process,
+                          addUTagInfo = nanoAOD_tau_switch.addPNet.value(),
+                          runUTagAK4 = nanoAOD_tau_switch.runUTagCHSAK4.value(),
+                          usePUPPIjets = False
+    )
+    if nanoAOD_tau_switch.addUParTInfo.value():
+        # Add Unified Tagger For PUPPI Jets (UParT 2024)
+        nanoAOD_addUTagToTaus(process,
+                            addUTagInfo = nanoAOD_tau_switch.addPNet.value(),
+                            runUTagAK4 = nanoAOD_tau_switch.runUTagPUPPIAK4.value(),
+                            usePUPPIjets = True
+        )
```

Comment:
I don't think the current implementation of `nanoAOD_addUTagToTaus()` allows adding both the CHS-based PNet and the Puppi-based UParT to the tau collection -- the second call would simply override the first call... 

Please clarify what is the intended behavior of the code and rework `nanoAOD_addUTagToTaus` if needed.